cupy/core/core.pyx

```diff
@@ -1865,6 +1865,7 @@ cdef list _cupy_header_list = [
     'cupy/complex.cuh',
     'cupy/carray.cuh',
     'cupy/atomics.cuh',
+    'cupy/tuple.cuh',
```

Comment:
Do we have to include this for every kernel?

Reply:
No, but it will not break compatibility.

Reply:
Initially I was only concerned about compilation time (which could be lengthened a bit), but now I am also concerned about potential conflict with Jitify:
https://github.com/NVIDIA/jitify/blob/60e9e7231fe7529ecfc76fa97b9c02b3033f4157/jitify.hpp#L2380
although it would only happen if `#include <tuple>` is done and `using std::tuple` is in use.